### ISSUE TITLE
[MacOS] IDE / GUI: Cursor "press" action is ignored

### ISSUE THREAD

**Issue by inhalt120g** [godotengine/godot#81539]
### Godot version

4.1.1 Stable

### System information

Mac OS Godot v4.1.1.stable - macOS 13.5.2 - Vulkan (Forward+) - integrated Apple M1 - Apple M1 (8 Threads)

### Issue description

On MacOS (and maybe other OSes), there is a particular L/R mouse button behavior behavior that is hard to live without under certain circumstances, I don't know whether it has an official name but let's call it "click and hold".

So, when the user presses and holds, for example, LMB on a menu, the selection highlight appears and stays on the menu as long as the button is kept pressed. When the button is released, whatever was under the highlight / cursor is selected. Note that this behavior **saves you basically two out of four finger movements** when compared to the standard "two clicks required for simple operations: one to open the menu, and then select something" where you have to perform 4 finger actions (finger press down, lift up, press down, lift up) and this way you can do it with only 2 (finger down, finger up).

This behavior is especially comfortable (kind to wrists) to use with graphic tablet if you're an artist for example, and use a pen instead of a mouse most of the time.

In Godot 4, the "click and hold" method works on all the top menus (Godot, Scene, Project, Debug Editor, Help) but not in the IDE: if I right click on, for example, a scene in the FileSystem panel and keep the RMB pressed, the highlight / selection does not appear (only the menu opens). So I have to click on the menu again just to select an entry. Always throws me off.

### Steps to reproduce

Open Godot.

Correct behavior:
Position your mouse over any top menu, for example "Scene".
Press and hold RMB (don't let go)!
Now drag the cursor a bit down, to, say, New Scene.
Release RMB.
A new scene will be created!
So it looks like this, note there's only one RMB press in the operation:
https://github.com/godotengine/godot/assets/106015575/ec932262-24f6-450f-a895-54632522cf32

Strange behavior:
Position your mouse over a scene in the FileSystem panel.
Press and hold RMB.
The menu opens, but there is no highlight marker!
Even if you wiggle the cursor out of the just opened menu area, nothing will happen.
Now you have to release RMB for the highlight marker to appear…
…and the click again.

### Minimal reproduction project

N/A

**Comment by YuriSizov** [comment]
Hey there! Please open a proposal about this, since it's a new feature that you are requesting. You can open a proposal in https://github.com/godotengine/godot-proposals.

**Comment by Calinou** [comment]
> In Godot 4, the "click and hold" method works on all the top menus (Godot, Scene, Project, Debug Editor, Help) but not in the IDE: if I right click on, for example, a scene in the FileSystem panel and keep the RMB pressed, the highlight / selection does not appear (only the menu opens). So I have to click on the menu again just to select an entry. Always throws me off.

This sounds like a bug. Can you reproduce this in 3.5.2?

**Comment by inhalt120g** [comment]
> Please open a proposal about this, since it's a new feature that you are requesting.

It's not a new feature, please see below. Also, if one part of the interface works under the "Amiga convention" and the other part doesn't (main menu and FileSystem menu respectively), surely that's a bug :P


> Can you reproduce this in 3.5.2?

In 3.5.2 it works in everything I tried (by "it works" I mean if I press and hold the mouse button, the menu highlight selection moves with it and when I release the mouse button the selection is activated automatically without me having to click one more time).


> reproducing this in 4.1.1 after enabling Single Window Mode in the Editor Settings then restarting the editor

When Single Window mode is enabled, the menus respect the "Amiga convention" (sorry I have no idea what is the official name for such behavior).

Thank you for your reply Calinou, it's really appreciated!

**Comment by YuriSizov** [comment]
> Also, if one part of the interface works under the "Amiga convention" and the other part doesn't (main menu and FileSystem menu respectively), surely that's a bug?

Not necessarily, no. If a feature was implemented in one place but wasn't implemented in others it's not a bug, just a missing feature. And it may require further justification than just "it works in one place, so it must work in other places". But if this is a regression since 3.x, then we can file it as a bug.
